src/Catalog/CatalogItem.cs

```diff
@@ -37,5 +51,23 @@ public virtual IGraph CreateContentGraph(CatalogContext context)
         {
             return null;
         }
+
+        /// <summary>
+        /// Changes the host of the baseAddress with the customDomain
+        /// </summary>
+        /// <param name="baseAddress"></param>
+        /// <param name="customDomain"></param>
+        /// <returns></returns>
+        public static  Uri AliasesAddress(Uri baseAddress, string customDomain)
```

Comment:
Given that you are using this method constantly outside of the context of the catalog, I would move this to another shared static class.